src/sys/inotify.rs

```diff
@@ -0,0 +1,225 @@
+//! Monitoring API for filesystem events.
+//!
+//! Documentation
+//! =============
+//!
+//! Inotify is a Linux-only API to monitor filesystems events.
+//!
+//! For more documentation, please read [inotify(7)](http://man7.org/linux/man-pages/man7/inotify.7.html).
+//! 
+//! Example
+//! =======
+//!
+//! * Monitor all events happening in directory "test":
+//! ```no_run
+//! # use nix::sys::inotify::{EventFlags,InotifyInitFlags,Inotify};
+//! #
+//! // We create a new inotify instance.
+//! let instance = Inotify::init(InotifyInitFlags::empty()).unwrap();
+//!
+//! // We add a new watch on directory "test" for all events.
+//! let wd = instance.add_watch("test", EventFlags::IN_ALL_EVENTS).unwrap();
+//!
+//! loop {
+//!     // We read from our inotify instance for events.
+//!     let events = instance.read_events().unwrap();
+//!     println!("Events: {:?}", events);
+//! }
+//! ```
+
+use libc;
+use libc::{
+    c_char,
+    c_int,
+    uint32_t
+};
+use std::ffi::{OsString,OsStr,CStr};
+use std::os::unix::ffi::OsStrExt;
+use std::mem::size_of;
+use std::os::unix::io::{RawFd,AsRawFd,FromRawFd};
+use unistd::read;
+use Result;
+use NixPath;
+use errno::Errno;
+
+libc_bitflags! {
+    /// Configuration options for [`inotify_add_watch`](fn.inotify_add_watch.html).
+    pub struct EventFlags: uint32_t {
+        IN_ACCESS;
+		IN_MODIFY;
+		IN_ATTRIB;
+		IN_CLOSE_WRITE;
+		IN_CLOSE_NOWRITE;
+		IN_OPEN;
+		IN_MOVED_FROM;
+		IN_MOVED_TO;
+		IN_CREATE;
+		IN_DELETE;
+		IN_DELETE_SELF;
+		IN_MOVE_SELF;
+
+		IN_UNMOUNT;
+		IN_Q_OVERFLOW;
+		IN_IGNORED;
+
+		IN_CLOSE;
+		IN_MOVE;
+
+		IN_ONLYDIR;
+		IN_DONT_FOLLOW;
+        
+		IN_ISDIR;
+		IN_ONESHOT;
+        IN_ALL_EVENTS;
+    }
+}
+
+libc_bitflags! {
+    /// Configuration options for [`inotify_init1`](fn.inotify_init1.html).
+    pub struct InotifyInitFlags: c_int {
+        IN_CLOEXEC;
+        IN_NONBLOCK;
+    }
+}
+
+/// An inotify instance. This is also a file descriptor, you can feed it to
+/// other interfaces consuming file descriptors, epoll for example.
+#[derive(Debug, Clone, Copy)]
+pub struct Inotify {
+    fd: RawFd
+}
+
+/// This object is returned when you create a new watch on an inotify instance.
+/// It is then returned as part of an event once triggered. It allows you to
+/// know which watch triggered which event. 
+#[derive(Debug, Clone, Copy, Hash, Eq, PartialEq, Ord, PartialOrd)]
+pub struct WatchDescriptor {
+    wd: i32
+}
+
+/// A single inotify event.
+/// For more documentation see, [inotify(7)](http://man7.org/linux/man-pages/man7/inotify.7.html).
```

Comment:
You should probably put a blank line between these two so the link will be hidden in the collapsed html documentation.